openedx/features/course_duration_limits/access.py

```diff
@@ -74,6 +74,36 @@ def get_user_course_expiration_date(user, course):
     if enrollment is None or enrollment.mode != CourseMode.AUDIT:
         return None
 
+    # The user course expiration date is the content availability date
+    # plus the weeks_to_complete field from course-discovery.
+    discovery_course_details = get_course_run_details(course.id, ['weeks_to_complete'])
+    expected_weeks = discovery_course_details.get('weeks_to_complete')
+    if expected_weeks:
+        access_duration = timedelta(weeks=expected_weeks)
+
+    # Course access duration is bounded by the min and max duration.
+    access_duration = max(MIN_DURATION, min(MAX_DURATION, access_duration))
+
+    return access_duration
+
+
+def get_user_course_expiration_date(user, course):
+    """
+    Return expiration date for given user course pair.
+    Return None if the course does not expire.
+
+    Business Logic:
+      - Course access duration is bounded by the min and max duration.
+      - If course fields are missing, default course access duration to MIN_DURATION.
+    """
+    access_duration = get_user_course_duration(user, course)
+    if access_duration is None:
+        return None
+
+    enrollment = CourseEnrollment.get_enrollment(user, course.id)
+    if enrollment is None or enrollment.mode != CourseMode.AUDIT:
+        return None
```

Comment:
why this new logic?  was this simply missed before?

Reply:
This is actually the same as it was before. The diff is strange because I moved a bunch of the code from this function into the new `get_user_course_duration` function, but these lines were actually duplicated (because I need the enrollment in both functions), so they look like they are new in this function.